src/integer/mat_z/arithmetic/div.rs

```diff
@@ -0,0 +1,153 @@
+// Copyright © 2023 Marcel Luca Schmidt
+//
+// This file is part of qFALL-math.
+//
+// qFALL-math is free software: you can redistribute it and/or modify it under
+// the terms of the Mozilla Public License Version 2.0 as published by the
+// Mozilla Foundation. See <https://mozilla.org/en-US/MPL/2.0/>.
+
+//! Implementation of scalar multiplication for [`MatZ`] matrices.
+
+use super::super::MatZ;
+use crate::integer::Z;
+use crate::macros::arithmetics::{
+    arithmetic_trait_borrowed_to_owned, arithmetic_trait_mixed_borrowed_owned,
+};
+use crate::traits::{GetNumColumns, GetNumRows};
+use flint_sys::fmpz_mat::fmpz_mat_scalar_divexact_fmpz;
+use std::ops::Div;
+
+impl Div<&Z> for &MatZ {
```

Comment:
Not using Generics, because that would be too much effort at the moment (Requires new macro).
Might be switched to generic when other functions are also updated.